### DISCUSSION TITLE
Can't import `org.oscim.layers.vector.geometries.Style`

### DISCUSSION THREAD

**Discussion by jhotadhari** [mapsforge/vtm#1139] [Q&A] [answered]
Hi,

I'm using version 0.22.0

trying to change the style of a `PathLayer`, but I can't import `org.oscim.layers.vector.geometries.Style`.
Android-Studio says *Cannot resolve symbol 'geometries'*

![image](https://github.com/user-attachments/assets/3558be35-0e73-4816-b14e-acf891a67a74)
https://github.com/jhotadhari/react-native-mapsforge-vtm/blob/6c6298a3b71defdd12fa1dd44f50e4f2857ad5b4/android/src/main/java/com/jhotadhari/reactnative/mapsforge/vtm/MapLayerPathModule.java

Is that some vtm bug or a misconfiguration in my project?

Thank you!!!

**Comment by devemux86** [comment, marked as answer]
The [Style](https://github.com/mapsforge/vtm/blob/master/vtm-jts/src/org/oscim/layers/vector/geometries/Style.java) class exists in the `vtm-jts` module.

See also the integration guide:
https://github.com/mapsforge/vtm/blob/master/docs/Integration.md#jts-overlays

**Reply by jhotadhari** [reply]
Thanks, these were the imports I forgot.

Furthermore I recognized, that there are two `PathLayer`, maybe that helps someone:  
- `org.oscim.layers.PathLayer` excepts `org.oscim.theme.styles.LineStyle`
- `org.oscim.layers.vector.PathLayer` excepts `org.oscim.layers.vector.geometries.Style`

**Reply by devemux86** [reply]
Use the `PathLayer` of the `vtm-jts` module, it works better.

**Reply by jhotadhari** [reply]
`PathLayer` of the `vtm-jts` module renders with some sharp edges.


```
import org.oscim.layers.vector.PathLayer;

...

			Style style = Style.builder()
				.strokeWidth(4)
				.strokeColor(Color.parseColor("#ff0000"))
				.build();
			PathLayer pathLayer = new PathLayer( mapView.map(), style );
```
looks like this 
![image](https://github.com/user-attachments/assets/da4a54e9-c321-4957-a726-8e5157826d33)


```
import org.oscim.layers.PathLayer

...

			LineStyle lineStyle = new LineStyle( Color.parseColor("#ff0000"), 4 );
			PathLayer pathLayer = new PathLayer( mapView.map(), lineStyle );
```
looks like this. renders nice even if I increase the strokeWidth a lot.
![image](https://github.com/user-attachments/assets/b5a405a8-f6d2-4aeb-bd1f-7609b315e629)

**Reply by devemux86** [reply]
Do not use a very large path layer width.

**Comment by victorlysak** [comment]


**Reply by devemux86** [reply]
@victorlysak Please create a new discussion to ask new questions.

You can study and test the [vtm-android-example](https://github.com/mapsforge/vtm/tree/master/vtm-android-example) samples to see how the library works.

The main page mentions:
"Building VTM: set `ANDROID_HOME` environment variable with the Android SDK installation directory"
